### ISSUE TITLE
Automatic Reconnect fails under certain circumstances

### ISSUE THREAD

**Issue by GiveMeMoreMtDew** [eclipse-paho/paho.mqtt.c#300]
Running three separate client applications and mosquitto broker.
Client application A and the broker are running on the same PC (Windows 10)
Client applications B and C are running on embedded devices.

Client application A creates four objects.  Each object is a separate MQTT client.  All objects on Client A connect to the broker using the URL "localhost:1883".  The automaticReconnect flag is set to 1.
Client applications B, C, and all objects on Client A connect successfully when the client applications are launched..
I terminate the broker using CTRL-C and restart the broker.
All clients detect the disconnect and call the connectionLost callback.
Clients B and C always reconnect.  
Very few, if any of the objects in Client A reconnect.  The log outputs show that the client is automatically reconnecting and repeats forever.

If i remove Clients B and C from the network, then the objects in Client A reconnect successfully.
If I use the URL "192.168.2.6:1883" (the PC's IP address) then the objects in Client A will reconnect when the embedded devices are present.

Obviously, I should not use localhost to define the location of the broker.  

Is this a known problem with PHO, mosquitto, or Windows 10?  Is there something I can do to make this scenario work and use localhost to define the broker location?

**Comment by icraggs** [comment]
Are there any useful messages on the Mosquitto log?

Taking a client trace of the failing connect attempt by setting the environment variable:

MQTT_C_CLIENT_TRACE=ON 

might give some clues.

**Comment by GiveMeMoreMtDew** [comment]
[log.txt](https://github.com/eclipse/paho.mqtt.c/files/1149279/log.txt)

The above file contains the PAHO trace.

FYI 
The PAHO DLL I am using has been compiled with Microsoft Visual Studio 2015.  The following flags are defined: NO_PERSISTENCE, NO_HEAP_TRACKING.

I have added some code (print statements) and comments to the files for debugging purposes, so the line numbers may not match what you have.  If the line numbers are important to you then I can always build the library using the original code and create a new trace file.

I have noticed that clients will reattach if the broker is restarted before the 5th "Automatically attempting to reconnect" message is displayed (line 1299 in the provided file).  Clients will not reattach if the broker is restarted after the 5th "reconnect" message.

The Mosquitto broker is also compiled with MSVS 2015 with the following flags defined:  WITH_BRIDGE, WITH_BROKER.  I was not aware that it created a log file.  I always run the broker with the "-v" option and I see clients come and go when things are working, but for these tests, I am terminating the broker, then restarting it.  If I wait to long to restart it, there are no connections.
